contracts/vault/LightAbstractVault.sol

```diff
@@ -39,6 +41,12 @@ abstract contract LightAbstractVault is IERC4626Vault, InitializableToken, Vault
         _asset = IERC20(_assetArg);
     }
 
+    /**
+     * @param _assetToBurn amount of assets that will be deposited and corresponding shares locked permanently
+     * @dev This is to prevent against loss of precision and frontrunning the user deposits by sandwitch attack. Should be a non-trivial amount.
+     */
+    function _burnOnCreate(uint256 _assetToBurn) internal virtual;
```

Comment:
Why not put in `_initialize` like the `AbstractVault`?

Reply:
Hmm, there was some logic which I can't remember now ><. Nevertheless, I see that we can do the `_initialize` so gonna make those changes.